### ISSUE TITLE
type parameters don't make sense

### ISSUE THREAD

**Issue by cyberhck** [apollographql/apollo-client#8537]
https://github.com/apollographql/apollo-client/blob/0e360efc3e0c0ecdb7ad3bc0c516dc6dbb48e126/src/core/watchQueryOptions.ts#L38

https://github.com/apollographql/apollo-client/blob/0e360efc3e0c0ecdb7ad3bc0c516dc6dbb48e126/src/core/watchQueryOptions.ts#L247-L250

https://github.com/apollographql/apollo-client/blob/0e360efc3e0c0ecdb7ad3bc0c516dc6dbb48e126/src/core/watchQueryOptions.ts#L141

Looking at these three snippets of code, for same type parameters, they have the position flipped only on Mutation type.

Is there any reasoning behind this?

however looking at:
https://github.com/apollographql/apollo-client/blob/0e360efc3e0c0ecdb7ad3bc0c516dc6dbb48e126/src/core/ApolloClient.ts#L350

https://github.com/apollographql/apollo-client/blob/0e360efc3e0c0ecdb7ad3bc0c516dc6dbb48e126/src/core/ApolloClient.ts#L321-L323

and
https://github.com/apollographql/apollo-client/blob/0e360efc3e0c0ecdb7ad3bc0c516dc6dbb48e126/src/core/ApolloClient.ts#L363-L365

If we could fix this, it'd help a lot on tooling, having similar API means easier code generation as well.

**Comment by cyberhck** [comment]
if I send a PR to align this, will that be accepted?

**Comment by benjamn** [comment]
@cyberhck What happened here is that some of these types originally had only `TVariables` (without `TData`) as a generic type parameter. Over time, we have added `TData` to most of those existing types, while also introducing new types that have both type parameters.

For new types, because there is no backwards compatibility obligation, we generally put `TData` first and `TVariables` second. However, we've stuck with the original ordering for existing types, because changing the order of type parameters (making `TVariables` no longer the first type parameter) would be a breaking change for anyone relying on passing these types explicitly (rather than letting them be type-inferred, which is more common).

In short, backwards compatibility seems more important than consistency here, though consistency going forward (`TData` then `TVars`) is also worthwhile.

**Comment by cyberhck** [comment]
thanks for the info, I'm closing this issue as it's asked and answered, feel free to reopen if you have any plans :slightly_smiling_face:
